src/glue/jobs/s3_to_json.py

```diff
@@ -0,0 +1,209 @@
+"""
+S3 to JSON job
+
+Bucket/sort JSON from a zip archive provided by Care Evolution into
+JSON datasets in S3. A JSON dataset is specific to a key prefix
+(for example: json/dataset=EnrolledParticipants/) and only contains
+files which share a similar schema.
+"""
+import datetime
+import io
+import os
+import json
+import logging
+import sys
+import zipfile
+import boto3
+from awsglue.utils import getResolvedOptions
+
+def write_file_to_json_dataset(z, json_path, dataset_identifier,
+        metadata, workflow_run_properties):
+    """
+    Write a JSON from a zipfile to a JSON dataset.
+
+    Metadata fields derived from the file basename are inserted as top-level fields
+    and the NDJSON is written to a JSON dataset in S3.
+
+    Args:
+        z (zipfile.Zipfile): The zip archive as provided by the data provider.
+        json_path (str): A JSON path relative to the root of `z`.
+        dataset_identifier (str): The data type of `json_path`.
+        metadata (dict): Metadata derived from the file basename.
+        workflow_run_properties (dict): The workflow arguments
+
+    Returns:
+        output_path (str) The local path the file was written to.
+    """
+    s3_client = boto3.client("s3")
+    logger = logging.getLogger(__name__)
+    os.makedirs(dataset_identifier, exist_ok=True)
+    s3_metadata = metadata.copy()
+    if s3_metadata["start_date"] is None:
```

Comment:
Does the `start_date` key always exist?

Reply:
No, not for data types which don't include a start date in their file name: https://sagebionetworks.jira.com/wiki/spaces/IBC/pages/2789376001/Notes#Types-of-data-included-in-archive

Reply:
Thanks - I'll give that page a read!  Thanks!